### PR TITLE
fixed learning rate, moved to new-style classes

### PR DESCRIPTION
**Pull request by msperber**
I hope the handling of the learning rate is correct according to the new DyNet conventions?

### CODE REVIEW

File: xnmt/xnmt_train.py [hunk 1]
```diff
@@ -112,11 +112,12 @@ def pack_batches(self):
 
   def dynet_trainer_for_args(self, args):
     if args.trainer.lower() == "sgd":
-      trainer = dy.SimpleSGDTrainer(model_globals.dynet_param_collection.param_col, learning_rate = args.learning_rate)
+      trainer = dy.SimpleSGDTrainer(model_globals.dynet_param_collection.param_col)
```
Comment (neubig) [inline xnmt/xnmt_train.py]: It's still fine to pass the learning rate through the constructor. Was this not working?

Comment (msperber) [inline xnmt/xnmt_train.py]: Hm, I got an error because learning_rate was not a known parameter. I did use the recent DyNet, but maybe there was still a versioning problem. In either case, if we're at it I guess we could also change this to support any trainer that's available in DyNet by using getattr()..
